base_layer/mmr/src/sparse_merkle_tree/tree.rs

```diff
@@ -22,6 +22,7 @@ use crate::sparse_merkle_tree::{
 pub enum UpdateResult {
     Updated(ValueHash),
     Inserted,
+    AlreadyExistingKey,
```

Comment:
`KeyAlreadyExists` or just `KeyExists`; and I'd move this to SMTError as well (see next comment)

Reply:
changed to error type